### DISCUSSION TITLE
Performance inquiry

### DISCUSSION THREAD

**Discussion by HexSleeves** [react-hook-form/react-hook-form#7536] [Q&A] [answered]
I built a custom form builder based on config objects that I hand it. I have been using form provider to stop the obsessive prop drilling but im curious the performance gains from passing props down vs. using the provider. It seems that useFormContext causes my component to rerender even when they shouldn't.

I use why did I render to check some of the causes and it stated that useFormContext causes my component to rerender due to form state changing yet, the objects are identical on each render. Is a deep comparison being used?

**Comment by Moshyfawn** [comment, marked as answer]
Check out [this RHF docs section on `FormProvider` performance](https://react-hook-form.com/advanced-usage/#FormProviderPerformance)
